adapters/llamacpp.py

```
@@ -0,0 +1,311 @@
+class LlamaCppAdapter(BaseCLIAdapter):
+    """Adapter for llama.cpp CLI tool (llama-cli) with auto-discovery."""
+
+    # Default search paths for GGUF model files
+    DEFAULT_SEARCH_PATHS = [
+        "~/.cache/llama.cpp/models",
+        "~/models",
+        "~/llama.cpp/models",
+        "/usr/local/share/llama.cpp/models",
+        "~/.ollama/models",  # Ollama's model directory
+        "~/.lmstudio/models",  # LM Studio's model directory
+    ]
```

Ruff (0.14.10)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)